src/lib.rs

```diff
@@ -241,17 +246,27 @@ extern "C" fn init(callbacks: *mut PluginCallbacks, config_path: *const c_char)
     match unsafe { callbacks.as_ref() } {
         Some(c) => {
             unsafe { CALLBACKS = Some(c) };
-            let (messages_tx, messages_rx) = mpsc::sync_channel(0);
-            MESSAGE_CHANNEL.set(messages_tx).expect("Big problem: message channel already initialized!");
-
-            thread::spawn(move || {
-                janus_verb!("Message processing thread is alive.");
-                for msg in messages_rx.iter() {
-                    if let Err(e) = handle_message_async(msg) {
-                        janus_err!("Error processing message: {}", e);
-                    }
-                }
-            });
+            let mut senders = Vec::new();
+            let cpus = num_cpus::get(); // Run one thread per logical CPU
```

Comment:
Perhaps ideally this would go in the config.

Reply:
Ah yeah, I had a config at first but then considered CPU to be ideal, but yeah as a default seems smart too.